vignettes/demo.Rmd

```diff
@@ -26,63 +28,36 @@ knitr::opts_chunk$set(
   collapse = TRUE,
   error = !in_pkgdown
 )
-Sys.setenv(GIT_AUTHOR_DATE = "2020-12-12 12:47:37Z")
-Sys.setenv(GIT_AUTHOR_NAME = "Kirill Müller")
-Sys.setenv(GIT_AUTHOR_EMAIL = "krlmlr+r@mailbox.org")
-Sys.setenv(GIT_COMMITTER_DATE = "2020-12-12 12:47:37Z")
-Sys.setenv(GIT_COMMITTER_NAME = "Kirill Müller")
-Sys.setenv(GIT_COMMITTER_EMAIL = "krlmlr+r@mailbox.org")
 ```
 
 ## Set up the development environment
 
-Before we start development for {SuperFrob}, we set up the basic development environment required for any typical R package.
+Before we start development for {tea}, we set up the basic development environment required for any typical R package.
 
 ### Create a package
 
 We will start by creating a new package.
 For this demo, the package is created in a temporary directory.
 A real project will live somewhere in your home directory.
 
-```{r tempdir}
-tempdir <- file.path(tempdir(), "fledge")
-dir.create(tempdir)
-```
-
 The `usethis::create_package()` function sets up a package project ready for development.
-The output shows the details of the package created.
 
 ```{r create-package, eval = FALSE}
-pkg <- usethis::create_package(file.path(tempdir, "SuperFrob"))
+pkg <- usethis::create_package("tea")
 ```
 
 ```{r create-package-real, echo = FALSE}
-pkg <- usethis::create_package(
-  file.path(tempdir, "SuperFrob"),
-  fields = list(Date = as.Date(Sys.getenv("GIT_COMMITTER_DATE")))
-)
+pkg_dir <- file.path(tempdir(check = TRUE), "fledge")
+dir.create(pkg_dir)
+pkg <- fledge::create_demo_project("tea", open = FALSE, dir = pkg_dir)
 ```
 
-In an interactive RStudio session, a new window opens.
+In an interactive RStudio session, a new window opens and you would work there.
 Users of other environments would change the working directory manually.
 For this demo, we manually set the active project.
 
-```{r set-focus, include = FALSE}
-wd <- getwd()
-
-knitr::knit_hooks$set(
-  pkg = function(before, options, envir) {
-    if (before) {
-      wd <<- setwd(pkg)
-    } else {
-      setwd(wd)
-    }
-
-    invisible()
-  }
-)
-
-knitr::opts_chunk$set(pkg = TRUE)
+```{r, change-dir, include=FALSE}
+knitr::opts_knit$set(root.dir = pkg)
```

Comment:
I had trouble deleting the temporary directory with the hook approach (for some reason I could not change the directory therefore I could not remove it!) hence the new approach based on https://bookdown.org/yihui/rmarkdown-cookbook/working-directory.html